mobile/test/common/integration/base_client_integration_test.h

```diff
@@ -41,15 +41,16 @@ class BaseClientIntegrationTest : public BaseIntegrationTest {
   BaseClientIntegrationTest(Network::Address::IpVersion ip_version,
                             const std::string& bootstrap_config = defaultConfig());
   virtual ~BaseClientIntegrationTest() = default;
+  // Note: This class does not inherit from testing::Test and so this TearDown() method
+  // does not override testing::Test::TearDown(). As a result, it will not be called
+  // automatically by gtest during shutdown and must be called manually.
+  void TearDown();
```

Comment:
this needs to be `virtual` since client_integration_test overrides it

Reply:
Confusingly, ClientIntegrationTest does not override it! Instead it overrides testing::Test::Teardown().